### ISSUE TITLE
Extract method refactoring produces an invalid code when Out Variable Declarations are involved

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#12122]
The refactoring is offered on an identifier in the Out Var Declaration. At some point it looked like it was also offered on the ```var``` keyword, but I couldn't repro that again. The outcome is similar with explicit type in the declaration.

Before:
```
        static void Main(string[] args)
        {
            if (M1(out var x2))
            {
                System.Console.WriteLine(x2);
            }
        }
```

After:
```
        static void Main(string[] args)
        {
            int x2;
            if (NewMethod(out x2))
            {
                System.Console.WriteLine(x2);
            }
        }

        private static bool NewMethod(out int x2)
        {
            return M1(out var x2);
        }
```

**Comment by CyrusNajmabadi** [comment]
Definitely a bug.  expected result should be something like:

```c#
static void Main(string[] args)
        {
            if (NewMethod(out var x2))
            {
                System.Console.WriteLine(x2);
            }
        }

        private static bool NewMethod(out int x2)
        {
            return M1(out var x2);
        }
```

**Comment by jcouv** [comment]
This was fixed by #13874. The generated code is now valid.

Open a new issue if you want to improve the generated code further (see comment below).

Before refactoring:
```C#
    static void Main(string[] args)
    {
        if (M1(out var x2))
        {
            System.Console.WriteLine(x2);
        }
    }
```
After extracting method for M1 invocation:
```C#
    static void Main(string[] args)
    {
        int x2; 
        if (NewMethod(out x2)) // this could still be improved to be NewMethod(out var x2)
        {
            System.Console.WriteLine(x2);
        }
    }

    private static bool NewMethod(out int x2)
    {
        return M1(out x2); // this is fixed, it used to be M1(out var x2)
    }
```
